cpp/Ice/optional/ContactDBI.cpp

```diff
@@ -21,15 +21,15 @@ ContactDBI::addContact(
     contact->name = name;
     if (type)
     {
-        contact->type = std::move(type);
+        contact->type = type;
     }
     if (number)
     {
-        contact->number = std::move(number);
+        contact->number = number;
```

Comment:
What was the problem with this `std::move` call?

Reply:
This type is trivially copyable. Using `std::move` doesn't do anything.

Reply:
I guess we want to do `std::move(*number)` to move the string?

Reply:
Removing for number was a mistake. I've added it back.